src/Routes.jsx

```diff
@@ -15,51 +15,95 @@ const routesLayer = new GeoJsonLayer({
   extruded: true,
 });
 
+// placed outside of component to use in both stop & vehicle layer
+const ICON_MAPPING = {
+  marker: {
+    x: 0, y: 0, width: 128, height: 128, mask: true,
+  },
+};
 
 class Routes extends Component {
+  getStopMarkers() {
+    /* returns new DeckGL Icon Layer displaying all stops on given routes */
+
+    // Push stop markers into data array
+    const data = this.props.state.routes.filter(r => r.stops).reduce((acc, curr) =>
+      acc.concat(curr.stops.reduce((a, c) => [...a, {
```

Comment:
so what we could do here instead of `acc.concat`, which mutates it, we could do
`return  [...acc, curr.stops.reduce(..., ]`, as that makes the reducer useful (otherwise we could've used a for-loop instead). If you were wondering why I prefer this over a for-loop where you're adding stuff to a list, it's because this makes it clear what the list has without having to trace through the code.

Reply:
the problem here is the depth. we have to iterate through all stops inside all routes (2 iteration). 
because the result of first iteration `[{}, {},...]` is array, if i do [...acc, curr.stops.reduce(.... ] result becomes [[{}, {},..], [{},..], [], ...] but we need [{},{},....]. 
I couldn't figure out a way to make it one iteration. we may do it in two seperate iteration.

Reply:
what about `return [...acc, ...curr.stops.reduce(..., ]`?

Reply:
(the `...` is the spread operator - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator

Reply:
oh, I see. perfect!
I will commit in a minute.